### PR TITLE
Fix incorrect Manufacturer's Data

### LINKED ISSUE #366: Manufacturer Data is not correct
I'm debugging with BlueSee and cross referencing when I port the implementation to a Flutter app using this library. I need a way to identify devices from certain manufacturers (not the specific device but a signature of the type of device so I know which model it is, etc). So I was going to use manufacturer data, but it's not working quite right.

- First, it's an `Int8List` and I would have expected a `Uint8List` so we can get the raw bytes and parse them.
- More concerning is that I just tested with 4 different devices, and for each, I'm getting the same Manufacturer Data (see below) across 4 difference devices when using this library but BlueSee is showing me very different data... so I think there is a bug here.

What I'm getting on all of them is this (even though BlueSee shows different info):
`[-103, -87, -18, 125, -89, 45, -70, -73, -85, 13, -85, 90]`

I'm testing on a physical device (`iPhone 11 Pro Max` running `iOS 13.3`).

### PR DESCRIPTION
**Pull request by mikolak**
Fixes #366

### CODE REVIEW

File: lib/internal/bridge/scanning_mixin.dart [hunk 1]
```diff
@@ -24,10 +28,6 @@ mixin ScanningMixin on FlutterBLE {
       },
     );
 
-    if (_scanEvents == null) {
```
Comment (dariuszseweryn) [inline lib/internal/bridge/scanning_mixin.dart]: is this related to the issue?

Comment (mikolak) [reply lib/internal/bridge/scanning_mixin.dart]: Reversed.

**Comment by mikolak** [comment]
@pawelByszewski @dariuszseweryn I've added tests.

File: test/scan_result.dart [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+            _createJsonScanResult(serviceData: serviceData);
+
+        ScanResult scanResult =
+            ScanResult.fromJson(jsonDecode(serializedScanResult), null);
+        expect(scanResult.advertisementData.serviceData, equals(serviceData));
+      });
+    }
+
+    testServiceDataDeserialization({
+      "1": Uint8List.fromList([0, 1, 2]),
```
Comment (pawelByszewski) [inline test/scan_result.dart]: for me, it is not readable at all, I dunno what `1` or `2` means

**Pull request merged**